lib/workers/bin/run_single_worker.rb

```diff
@@ -94,33 +94,30 @@ def all_role_names
   exit 1
 end
 
+options[:system_uid] ||= worker_class.system_uid(options[:guid]) if options[:guid]
+
 worker_class.preload_for_worker_role if worker_class.respond_to?(:preload_for_worker_role)
 unless options[:dry_run]
-  create_options = {:pid => Process.pid}
+  worker_options = {:pid => Process.pid}
   runner_options = {}
 
-  create_options[:system_uid] = options[:system_uid] if options[:system_uid]
+  # If we're given a GUID (systemd) then build the system_uid, otherwise (podified)
+  # we're given a system_uid already
+  options[:system_uid]      ||= worker_class.system_uid(options[:guid]) if options[:guid]
```

Comment:
Will this ever run if it's set on line 97?

Reply:
Ah just a bad rebase/conflict fix I think, thanks